Source/WebKit/UIProcess/API/C/WKRemoteInspectorServer.cpp

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (C) 2023 Sony Interactive Entertainment Inc.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WKRemoteInspectorServer.h"
+
+#if ENABLE(REMOTE_INSPECTOR)
+#include <JavaScriptCore/RemoteInspectorServer.h>
+
+uint16_t WKRemoteInspectorServerStart(const char* address, uint16_t port)
```

Comment:
This code looks exactly same with `JSRemoteInspectorServerStart`. What will happend if you call `JSRemoteInspectorServerStart` instead of `WKRemoteInspectorServerStart`?
https://github.com/WebKit/WebKit/blob/23bae944192123a739e1519b96081ec235e73800/Source/JavaScriptCore/API/JSRemoteInspectorServer.cpp#L32-L39

BTW, I think we should add a new API `JSRemoteInspectorServerStop` to stop the server.

Reply:
>  What will happend if you call JSRemoteInspectorServerStart instead of WKRemoteInspectorServerStart?

The results will be exactly the same.

Reply:
If so, do we need the new API?

Reply:
That reminds me.
For some reason, I had assumed that the JS_EXPORT functions should not be called from the MiniBrowser.
Just to confirm, what you are saying is that we can call `JSRemoteInspectorServerStart` directly here, is that correct?

Reply:
> That reminds me. For some reason, I had assumed that the JS_EXPORT functions should not be called from the MiniBrowser. 

Why?

> Just to confirm, what you are saying is that we can call `JSRemoteInspectorServerStart` directly here, is that correct?

No.
I'd like to know how do you think about an idea using the existing API instead of adding same API.

Reply:
> Why?

As I said above, "for some reason". @fujii -san's comment helped me realize that was a wrong thought.

> I'd like to know how do you think about an idea using the existing API instead of adding same API.

Since RemoteInspectorServer is part of JSC, I think it is appropriate to start it via `JSRemoteInspectorServerStart()`.
If it's not a problem, I'm willing to change it that way, but would appreciate your opinion.